dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp

```diff
@@ -106,47 +94,53 @@ ColumnDefine getColumnDefineForColumnExpr(const tipb::Expr & expr, const ColumnD
     return columns_to_read[column_index];
 }
 
-enum class OperandType
-{
-    Unknown = 0,
-    Column,
-    Literal,
-};
-
 inline RSOperatorPtr parseTiCompareExpr( //
     const tipb::Expr & expr,
     const FilterParser::RSFilterType filter_type,
     const ColumnDefines & columns_to_read,
     const FilterParser::AttrCreatorByColumnID & creator,
-    const TimezoneInfo & timezone_info,
-    const LoggerPtr & /*log*/)
+    const TimezoneInfo & timezone_info)
 {
-    if (unlikely(
-            expr.children_size() != 2 && filter_type != FilterParser::RSFilterType::In
-            && filter_type != FilterParser::RSFilterType::NotIn))
+    if (unlikely(expr.children_size() != 2 && filter_type != FilterParser::RSFilterType::In))
         return createUnsupported(
             expr.ShortDebugString(),
-            tipb::ScalarFuncSig_Name(expr.sig()) + " with " + DB::toString(expr.children_size())
-                + " children is not supported",
+            fmt::format(
+                "{} with {} children is not supported",
+                tipb::ScalarFuncSig_Name(expr.sig()),
+                expr.children_size()),
             false);
 
-    /// Only support `column` `op` `literal`, and `column` in (literal1, literal2, ...) now.
+    // Support three types of expression:
+    // 1. `column` `op` `literal`
+    // 2. `literal` `op` `column`
+    // 3. `column` in (literal1, literal2, ...)
 
     Attr attr;
     std::vector<Field> values;
-    OperandType left = OperandType::Unknown;
-    OperandType right = OperandType::Unknown;
     bool is_timestamp_column = false;
     for (const auto & child : expr.children())
     {
         if (isColumnExpr(child))
+        {
             is_timestamp_column = (child.field_type().tp() == TiDB::TypeTimestamp);
+            break;
+        }
     }
-    for (int32_t child_idx = 0; child_idx < expr.children_size(); ++child_idx)
+    int column_child_idx = -1;
+    for (int child_idx = 0; child_idx < expr.children_size(); ++child_idx)
     {
         const auto & child = expr.children(child_idx);
         if (isColumnExpr(child))
         {
+            // If iter to the second column, return unsupported.
+            if (child_idx == 0 || column_child_idx == -1)
+                column_child_idx = child_idx;
+            else
+                return createUnsupported(
+                    expr.ShortDebugString(),
+                    "Multiple ColumnRef in expression is not supported",
+                    false);
```

Comment:
Why we need `child_idx == 0 || column_child_idex == -1` here? Is it OK to only check `column_child_index == -1`?

Reply:
Fixed in https://github.com/pingcap/tiflash/pull/8212/commits/b38399f882107aa97032c04e04713a4166aa0bd3